app/components/DeployContract.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState } from 'react';
+import { useAppContext } from '../hooks/useAppContext';
+import { useWalletClient, useChainId, useSwitchChain, useAccount } from 'wagmi';
+import { createPublicClient, http, Chain, Hash, encodeFunctionData } from 'viem';
+import { 
+  base,
+  baseSepolia,
+  sepolia,
+  polygon,
+  arbitrum,
+  optimism,
+  zora,
+  mainnet,
+  localhost,
+} from 'viem/chains';
+import { SolidityFile } from '../types';
+import { 
+  ConnectWallet, 
+  Wallet, 
+  WalletDropdown, 
+  WalletDropdownLink, 
+  WalletDropdownDisconnect, 
+} from '@coinbase/onchainkit/wallet'; 
+import {
+  Address,
+  Avatar,
+  Name,
+  Identity,
+  EthBalance,
+} from '@coinbase/onchainkit/identity';
+import { color } from '@coinbase/onchainkit/theme';
+
+const chains: Chain[] = [ // TODO: this is defined in multiple places
+  mainnet,
+  sepolia,
+  polygon,
+  arbitrum,
+  optimism,
+  zora,
+  base,
+  baseSepolia,
+  localhost,
+];
+
+interface DeployContractProps {
+  onClose: () => void;
+}
+
+const DeployContract: React.FC<DeployContractProps> = ({ onClose }) => {
+  const { currentFile, currentFileCompilationResult, addNewContract } = useAppContext();
+  const { data: walletClient } = useWalletClient();
+  const { switchChain } = useSwitchChain();
+  const chainId = useChainId();
+  const { address, isConnected } = useAccount();
+  
+  const [isDeploying, setIsDeploying] = useState(false);
+  const [selectedChain, setSelectedChain] = useState(base);
+  const [error, setError] = useState<string | null>(null);
+
+  const deployContract = async () => {
+    if (!isConnected || !address) {
+      setError('Please connect your wallet first');
+      return;
+    }
+    
+    if (!walletClient) {
+      setError('Please connect your wallet first');
+      return;
+    }
+
+    setIsDeploying(true);
+    setError(null);
+    if (!currentFile || !currentFileCompilationResult) {
+      setError('No contract selected or compilation result not available');
+      return;
+    }
+
+    setIsDeploying(true);
+    setError(null);
+
+    try {
+      // Ensure the connected chain matches the selected chain
+      console.log('Starting deployment process');
+      console.log('Current chain:', chainId);
+      console.log('Selected chain:', selectedChain.id);
+
+      if (chainId !== selectedChain.id) {
+        console.log('Switching chain');
+        await switchChain({ chainId: selectedChain.id });
+      }
+
+      console.log('Preparing contract deployment');
+
+      const publicClient = createPublicClient({
+        chain: selectedChain,
+        transport: http()
+      });
+
+      const bytecode = currentFileCompilationResult.evm.bytecode.object as `0x${string}`;
+      console.log('ABI:', currentFileCompilationResult.abi);
+      console.log('Bytecode:', bytecode);
+      console.log('Account:', address);
+
+      const hash = await walletClient.deployContract({
```

Comment:
I think I would use sendCalls here where the `to` address is the safe factory and the calldata is https://github.com/wilsoncusack/safe-singleton-deployer-sol/blob/main/src/SafeSingletonDeployer.sol#L76